### ISSUE TITLE
Get Event parameter for onChange event

### ISSUE THREAD

**Issue by kirankuh** [ericgio/react-bootstrap-typeahead#701]
I'm in a situation where I need to persist my event using event.persist() but unfortunately onChange does not prove any event, it just gives selected values and I don't want to use onInputChange because it renders for every keystroke so I need a final selected value and onChange provides that but without event.

**Comment by ericgio** [comment]
Hi @kirankuh, thanks for the feature request. Can you describe how you're using `onChange` and why you need the event?

For context, the `onChange` hook isn't a true dom event, which is why none is provided. There's an event that _triggers_ the change, like a click or a keypress, but otherwise the hook is just signaling an internal state change.

**Comment by kirankuh** [comment]
Hi Ericgio,

My use case was I wanted to retain the user selected autocomplete value so
initially I wanted to fit typeahed with existing logic

We were using HTML dropdown by default and we were able to retain the value
when user is pressed submit button and comes back so that they don't need
to fill again

We were using refs to retain value and event persist but I see typeahead
doesn't have that synthetic event so retaining the value was getting
harder.

So what I have tried with typeahed is
1. Oninputchange will not trigger when selected it only triggers when input
is found so not helpful for my case
2. Typeahead had some prop called defaultinput something like that so I
used this but it didn't work as expected i.e it doesn't show value even
after setting the state
3. Typeahead doesn't have name prop which is very useful in many cases and
I don't see any docs re how to add our own props to the component


Tldr; I wanted to retain the selected value when user comes back to the
page after submitting

Thanks,
Kiran H

**Comment by ericgio** [comment]
`onChange` provides an array of the selected value(s) as the argument. You shouldn't need the event.
